bootstrapvz/common/tasks/cleanup.py

```diff
@@ -30,3 +30,15 @@ def run(cls, info):
         log = os.path.join(info.root, 'var/log/')
         os.remove(os.path.join(log, 'bootstrap.log'))
         os.remove(os.path.join(log, 'dpkg.log'))
+
+
+class ClearMACHINEID(Task):
+    description = 'Clearing the Machine ID'
+    phase = phases.system_cleaning
+
+    @classmethod
+    def run(cls, info):
+        for machineid_file in [os.path.join(info.root, f) for f in ['etc/machine-id', 'var/lib/dbus/machine-id']]:
+            if os.path.isfile(machineid_file):
```

Comment:
Check out the "[Roll complexity into which tasks are added to the tasklist](http://bootstrap-vz.readthedocs.io/en/master/developers/contributing.html#roll-complexity-into-which-tasks-are-added-to-the-tasklist)" chapter in the docs. If the file does not exist to begin with the task shouldn't be added to the list during resolution, otherwise you cannot rely on the tasklog to tell you what work was done to create an image.
To do that, I suggest changing `cleanup_group` into a function that checks the release that is being bootstrapped for and conditionally adds the task.

Reply:
You're right. I was thinking about it and here are some thoughts:
* ```/etc/machine-id``` may not exist if someone installs sysvinit instead of systemd regardless of Debian release.
* ```/var/lib/dbus/machine-id``` is a file from dbus package (the dbus binary creates it, not package or dpkg scripts). dbus package is not installed by default, but someone may install it via ```include_packages```, and IMO it should be cleared even in this case. 

So both files may or may not exist regardless of the release.  Just the probability is different (like 0% for wheezy and 98% for stretch). 

I don't know how to proceed in that case. WDYT @andsens ?

Reply:
OK, I would only add it for stretch then and add a debug message to the task if the file does not exist, that's the best you can do really.